src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -228,6 +242,30 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
             return base.UpdateValueCore(updateValueParams);
         }
 
+        /// <summary>
+        /// Updates the Script property of the node.
+        /// NOTE: This is a temporary method used during the Python 2 to Python 3 transistion period,
+        /// it will be removed when the transistion period is over.
+        /// </summary>
+        /// <param name="newCode">The new migrated code</param>
+        public void MigrateCode(string newCode)
+        {        
+            var e = new PythonCodeMigrationEventArgs(Script, newCode);
+            OnCodeMigrated(e);
```

Comment:
always fire the `something changed/happened` event after saving the changes, so move to after the `Script = newCode` line